### ISSUE TITLE
etcd stuck in initializing state [race condition when looking for current service scale and using scale_policy]

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#5333]
**Rancher Version:**: 1.1

**Docker Version:**1.10.3

**OS and where are the hosts located? (cloud, bare metal, etc):**: Ubuntu, Cloud

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**: Kubernetes

**Steps to Reproduce:**
1. Create a Kubernetes set up with 3 hosts. Use qa catalog. Added the 3 hosts through digital ocean

**Results:** 
 etcd is stuck in intializing state

```
+ '[' 1 -eq 0 ']'
+ eval node
++ node
++ '[' 2 == 1 ']'
++ '[' -d /data/member ']'
+++ etcdctln member list
+++ target=0
+++ grep etcd1
++++ seq 1 5
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ '[' 0 == 0 ']'
+++ echo No etcd nodes available
++ '[' '' '!=' '' ']'
+++ etcdctln get _state
+++ target=0
++++ seq 1 5
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ for j in '$(seq 1 5)'
++++ seq 1 2
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_1:2379/health
+++ '[' 1 == 0 ']'
+++ for i in '$(seq 1 $SCALE)'
+++ giddyup probe http://Kubernetes_etcd_2:2379/health
+++ '[' 1 == 0 ']'
+++ '[' 0 '!=' 0 ']'
+++ sleep 1
+++ '[' 0 == 0 ']'
+++ echo No etcd nodes available
++ '[' 'No etcd nodes available' '!=' RUNNING ']'
++ '[' 1 == 1 ']'
++ bootstrap_node
++ echo Waiting for discovery node to become healthy
Waiting for discovery node to become healthy
++ giddyup probe http://discovery:6666/health --loop --min 1s --max 60s --backoff 1.1
Get http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
Get http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
Get http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host


```

**Comment by LLParse** [comment]
The client-side solution to this issue is to not rely on current scale for critical decisions. In this specific case, we can instead use min scale to determine if bootstrapping is necessary.

**Comment by soumyalj** [comment]
Still see the same issue: Started a cattle environment with 3 hosts from GCE.  Added the QA catalog and changed cattle to Kubernetes. etcd is stuck in initializing state.

7/7/2016 1:18:52 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:52 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:52 PM+++ giddyup probe http://Kubernetes_etcd_2:2379/health
7/7/2016 1:18:52 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:52 PM+++ '[' 0 '!=' 0 ']'
7/7/2016 1:18:52 PM+++ sleep 1
7/7/2016 1:18:53 PM+++ for j in '$(seq 1 5)'
7/7/2016 1:18:53 PM++++ seq 1 2
7/7/2016 1:18:53 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:53 PM+++ giddyup probe http://Kubernetes_etcd_1:2379/health
7/7/2016 1:18:53 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:53 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:53 PM+++ giddyup probe http://Kubernetes_etcd_2:2379/health
7/7/2016 1:18:53 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:53 PM+++ '[' 0 '!=' 0 ']'
7/7/2016 1:18:53 PM+++ sleep 1
7/7/2016 1:18:54 PM+++ for j in '$(seq 1 5)'
7/7/2016 1:18:54 PM++++ seq 1 2
7/7/2016 1:18:54 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:54 PM+++ giddyup probe http://Kubernetes_etcd_1:2379/health
7/7/2016 1:18:54 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:54 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:54 PM+++ giddyup probe http://Kubernetes_etcd_2:2379/health
7/7/2016 1:18:54 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:54 PM+++ '[' 0 '!=' 0 ']'
7/7/2016 1:18:54 PM+++ sleep 1
7/7/2016 1:18:55 PM+++ for j in '$(seq 1 5)'
7/7/2016 1:18:55 PM++++ seq 1 2
7/7/2016 1:18:55 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:55 PM+++ giddyup probe http://Kubernetes_etcd_1:2379/health
7/7/2016 1:18:55 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:55 PM+++ for i in '$(seq 1 $SCALE)'
7/7/2016 1:18:55 PM+++ giddyup probe http://Kubernetes_etcd_2:2379/health
7/7/2016 1:18:55 PM+++ '[' 1 == 0 ']'
7/7/2016 1:18:55 PM+++ '[' 0 '!=' 0 ']'
7/7/2016 1:18:55 PM+++ sleep 1
7/7/2016 1:18:56 PM+++ '[' 0 == 0 ']'
7/7/2016 1:18:56 PM+++ echo No etcd nodes available
7/7/2016 1:18:56 PMWaiting for discovery node to become healthy
7/7/2016 1:18:56 PM++ '[' 'No etcd nodes available' '!=' RUNNING ']'
7/7/2016 1:18:56 PM++ '[' 1 == 1 ']'
7/7/2016 1:18:56 PM++ bootstrap_node
7/7/2016 1:18:56 PM++ echo Waiting for discovery node to become healthy
7/7/2016 1:18:56 PM++ giddyup probe http://discovery:6666/health --loop --min 1s --max 60s --backoff 1.1
7/7/2016 1:18:56 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:18:57 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:18:59 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:00 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:01 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:03 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:04 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:06 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:08 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:10 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:13 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:15 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:18 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:21 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:25 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:28 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:33 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:37 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:42 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:48 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:19:54 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:01 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:08 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:16 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:25 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:35 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:46 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:20:58 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:21:11 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:21:25 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:21:41 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:21:59 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:22:18 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:22:39 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:23:02 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:23:28 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:23:56 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:24:27 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:25:01 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:25:38 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:26:19 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:27:05 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:27:54 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:28:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:29:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:30:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:31:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:32:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:33:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:34:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:35:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host
7/7/2016 1:36:49 PMGet http://discovery:6666/health: dial tcp: lookup discovery on 169.254.169.250:53: no such host

**Comment by LLParse** [comment]
Solved in Rancher `v1.1.1` and latest k8s template
